jep/0000/README.adoc

```diff
@@ -0,0 +1,219 @@
+That trick relied on a custom patch to XStream which has no upstream equivalent.
+Therefore, `ToolInstallation` is no longer `Serializable` as far as Java (or Pipeline) serialization is concerned.
+
+[source,bash]
+----
+find *-plugin/{,*/}src/main/java -type f -print0 | xargs -0 egrep -l 'class \S+ extends ToolInstallation' | xargs fgrep -l MasterToSlave
+----
+
+can be used to find plugins which might be serializing `ToolInstallation` over Remoting.
+These should stop doing so (usually refactoring the remote callable to be `static` and carry just a `String home` field),
+but will remain compatible for now with a runtime warning.
+
+=== Inherited private serialization methods
+
+Older versions of XStream honored `readResolve` or `writeReplace` methods in subclasses even when declared as `private` in the superclass.
+This violated the expectations of Java serialization, and so this behavior has now been fixed:
+only accessible (e.g., `protected`) methods of these names will be considered in the inheritance chain.
+
+The `TestBuilder` test utility in `jenkins-test-harness`, among other classes,
+relied on this trick to avoid saving itself to `config.xml`
+(which would frequently fail due to captured outer class references).
+Numerous plugins use `TestBuilder` (or, more rarely, `TestNotifier`) in functional tests.
+These tests should still run, even if the build step is an inner class,
+but will produce a runtime warning; it is better to update the parent POM to 4.8 or newer to pick up a fix.
+
+=== `XStreamSerializable`
+
+The custom annotation `XStreamSerializable` is no longer available.
+There is no equivalent.
+It was used only in `ToolInstaller`.
+
+=== `XStreamSerializeAs`
+
+The custom annotation `XStreamSerializeAs` is no longer available.
+`XStreamAliasType` can be used instead.
+It was used only in the `persona` plugin,
+which has since been link:https://www.jenkins.io/security/advisory/2020-10-08/#SECURITY-2046[removed from the update center].
+
+=== `HierarchicalStreamReader.peekNextChild`
+
+This method no longer exists.
+It has long been defined in a subtype `ExtendedHierarchicalStreamReader`;
+only the fork redundantly defined it also in the supertype.
+
+== Security
+
+There are no known security risks related to this proposal.
+Defenses introduced in JEP-200 are left intact,
+even though newer versions of XStream include their own simpler serialization security system.
+
+== Infrastructure Requirements
+
+There are no new infrastructure requirements related to this proposal.
+
+== Testing
+
+Besides tests inside Jenkins core itself,
+CloudBees will endeavor to verify that all
+link:https://docs.cloudbees.com/search?&type=ci-plugins&ci-plugins-tier=verified[“Tier 1”] and
+link:https://docs.cloudbees.com/search?&type=ci-plugins&ci-plugins-tier=compatible[“Tier 2”]
+plugins are compatible with the core changes,
+as determined by acceptance tests (ATH) and `plugin-compat-tester` (PCT).
```

Comment:
Needs a community sign-off before merging.
Not a subject for review in the case of JEP Draft acceptance process

Reply:
Sign off on what exactly?